libs/server/Storage/Session/ObjectStore/SortedSetOps.cs

```diff
@@ -561,45 +561,116 @@ public unsafe GarnetStatus SortedSetDifference(ArgSlice[] keys, out Dictionary<b
+                createTransaction = true;
+                txnManager.SaveKeyEntryToLock(destinationKey, true, LockType.Exclusive);
+                foreach (var item in keys)
+                    txnManager.SaveKeyEntryToLock(item, true, LockType.Shared);
+                _ = txnManager.Run(true);
+            }
+
+            var objectContext = txnManager.ObjectStoreLockableContext;
+
+            try
+            {
+                var status = SortedSetDifference(keys, ref objectContext, out var pairs);
+
+                if (status != GarnetStatus.OK)
                 {
-                    if (firstObj.garnetObject is not SortedSetObject firstSortedSet)
-                    {
-                        return GarnetStatus.WRONGTYPE;
-                    }
+                    return GarnetStatus.WRONGTYPE;
+                }
 
-                    if (keys.Length == 1)
-                    {
-                        pairs = firstSortedSet.Dictionary;
-                        return GarnetStatus.OK;
-                    }
+                count = pairs is null ? 0 : pairs.Count;
 
-                    // read the rest of the keys
-                    for (var item = 1; item < keys.Length; item++)
+                if (count > 0)
+                {
+                    SortedSetObject newSetObject = new();
+                    foreach (var (element, score) in pairs)
                     {
-                        statusOp = GET(keys[item].ToArray(), out var nextObj, ref objectContext);
-                        if (statusOp != GarnetStatus.OK)
-                            continue;
-
-                        if (nextObj.garnetObject is not SortedSetObject nextSortedSet)
-                        {
-                            pairs = default;
-                            return GarnetStatus.WRONGTYPE;
-                        }
-
-                        if (pairs == default)
-                            pairs = SortedSetObject.CopyDiff(firstSortedSet.Dictionary, nextSortedSet.Dictionary);
-                        else
-                            SortedSetObject.InPlaceDiff(pairs, nextSortedSet.Dictionary);
+                        newSetObject.Add(element, score);
                     }
+                    _ = SET(destinationKey.ToArray(), newSetObject, ref objectContext);
                 }
+                else
+                {
+                    _ = EXPIRE(destinationKey, TimeSpan.Zero, out _, StoreType.Object, ExpireOption.None,
+                        ref lockableContext, ref objectContext);
+                }
+
+                return status;
             }
             finally
             {
                 if (createTransaction)
                     txnManager.Commit(true);
             }
+        }
+
+        private unsafe GarnetStatus SortedSetDifference<TObjectContext>(ReadOnlySpan<ArgSlice> keys, ref TObjectContext objectContext, out Dictionary<byte[], double> pairs)
```

Comment:
Please move to under all public methods in this file

Reply:
Done